### ISSUE TITLE
Site:GroundTemperature:Undisturbed:FiniteDifference initialization

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#6928]
Issue overview
--------------
From [here](https://github.com/NREL/EnergyPlus/pull/6478/commits/7042b1ef3abd464fc2328959e0ff03e40324a871#r212616314) in #6478.

In GroundTemperatureModeling/FiniteDifferenceGroundTemperatureModel::FiniteDiffGroundTempsModel::getWeatherData, the calculation for `annualAveAirTemp` seems suspect. 

he average temperature calc [here](https://github.com/NREL/EnergyPlus/blob/c4de98c52a99dccf8ef85e34b9a50a5fbda0f0aa/src/EnergyPlus/GroundTemperatureModeling/FiniteDifferenceGroundTemperatureModel.cc#L350) is dividing by `NumDaysInYear`. 

At first, it seems like it should be dividing by `DayOfSim`. But even that is incorrect, because this calculation is happening outside the environment loop, so `annualAveAirTemp_num` has potentially been accumulated for multiple environments. Oh, but this is creating and then deleting a temporary environment? I guess I don't fully understand what this function is trying to do.

Note that I made a last-minute change [here](https://github.com/NREL/EnergyPlus/pull/6478/commits/7042b1ef3abd464fc2328959e0ff03e40324a871) to get past a test failure, but that should probably be reverted and perhaps the problem is [here](https://github.com/NREL/EnergyPlus/blob/c4de98c52a99dccf8ef85e34b9a50a5fbda0f0aa/src/EnergyPlus/GroundTemperatureModeling/FiniteDifferenceGroundTemperatureModel.cc#L227-L236) instead:
```
 RunPeriodInput(TotRunPers).StartDate = JulianDay(1, 1, 0);
 RunPeriodInput(TotRunPers).EndDate = JulianDay(12, 31, 0);
```
@jasondegraw 

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by jasondegraw** [comment]
@mjwitte @mitchute It looks to me like this function is doing a pseudo-simulation by inserting an environment, iterating through the days and collecting weather stats, then removing the environment. Was `ksReadAllWeatherData` added for this one case?

I'm not sure how any date calculation could have ended up with 366 days in there for this year, seeing if I can reproduce now.

**Comment by mjwitte** [comment]
This should probably be consolidated with the similar `WeatherManager::AnnualMonthlyDryBulbWeatherData::CalcAnnualAndMonthlyDryBulbTemp()` function added [here](https://github.com/NREL/EnergyPlus/pull/6771/files#diff-3db4af63e54cde7a34f2fab72ff096fdR10319) in PR#6771.

**Comment by jasondegraw** [comment]
@mjwitte Are you sure it was only 366? I'm seeing *much* worse. The issue is indeed at the location of the two lines you called out in your comment above. The RunPeriod PR shifted a lot more of the logic into the setup of the RunPeriod object, and those two lines just aren't enough now. This is fixed on my new branch by providing better initialization in the ctor.

**Comment by mjwitte** [comment]
@jasondegraw Can't say completely for sure. The arrays bounds tripped when it tried to loop into day 366, so I'm not sure if I checked the value of `NumOfDayInEnvrn` or just assumed it was 366. I could go back and reproduce if that's helpful.

**Comment by jasondegraw** [comment]
@mjwitte No need to go back, cracking up on 366 makes sense (it's one over an array size). The actual value I saw was over 700, putting in a better end year than 0 fixes it and it should be 365 now.

**Comment by mjwitte** [comment]
Closed via #6941
